### ISSUE TITLE
After updating to 3.0.23 editing matrix field throws twig error - SuperTable

### ISSUE THREAD

**Issue by sergeifilippov** [craftcms/cms#3269]
### Description
After updating to 3.0.23, when you go to edit Matrix field settings a twig error is thrown.

### Steps to reproduce

1. Update to 3.0.23
2. In CP, navigate to editing matrix field settings. `settings/fields/edit/11`

### Additional info

- Craft version: 3.0.23
- PHP version: 7.1.21
- Database driver & version: Mysql 5.7
- Plugins & versions: SEOmatic, Imager, defaultab, freeform

**Comment by brandonkelly** [comment]
@sergeifilippov What is the error?

**Comment by sergeifilippov** [comment]
Sorry forgot to add it.

I've tracked it down to **Supertable** plugin.

But the error was:

```
Twig_Error_Runtime: Variable "blockTypes" does not exist. in src/vendor/craftcms/cms/src/templates/_components/fieldtypes/Matrix/settings.html:12
Stack trace:
#0 src/storage/runtime/compiled_templates/fe/fe5a8c1c724c74f0cd80a327fcb75920c63cd5ede02bb811d0ce4d401321736f.php(44): __TwigTemplate_e4d85ed2fca4f5927f35b4480f0791ef684c62f5ea200d6796558e6a72aef14c->{closure}()
#1 src/vendor/twig/twig/lib/Twig/Template.php(390): __TwigTemplate_e4d85ed2fca4f5927f35b4480f0791ef684c62f5ea200d6796558e6a72aef14c->doDisplay(Array, Array)
#2 src/vendor/craftcms/cms/src/web/twig/Template.php(49): Twig_Template->displayWithErrorHandling(Array, Array)
#3 src/vendor/twig/twig/lib/Twig/Template.php(367): craft\web\twig\Template->displayWithErrorHandling(Array, Array)
#4 src/vendor/craftcms/cms/src/web/twig/Template.php(31): Twig_Template->display(Array, Array)
#5 src/vendor/twig/twig/lib/Twig/Template.php(375): craft\web\twig\Template->display(Array)
#6 src/vendor/twig/twig/lib/Twig/Environment.php(289): Twig_Template->render(Array)
#7 src/vendor/craftcms/cms/src/web/View.php(331): Twig_Environment->render('_components/fie...', Array)
#8 src/vendor/verbb/super-table/src/services/SuperTableMatrixService.php(125): craft\web\View->renderTemplate('_components/fie...', Array)
#9 src/vendor/verbb/super-table/src/fields/SuperTableField.php(740): verbb\supertable\services\SuperTableMatrixService->getMatrixSettingsHtml(Object(craft\fields\Matrix))
#10 src/vendor/verbb/super-table/src/fields/SuperTableField.php(267): verbb\supertable\fields\SuperTableField->_getFieldOptionsForConfigurator()
#11 src/vendor/craftcms/cms/src/fields/Matrix.php(755): verbb\supertable\fields\SuperTableField->getSettingsHtml()
#12 src/vendor/craftcms/cms/src/fields/Matrix.php(265): craft\fields\Matrix->_getFieldOptionsForConfigurator()
#13 src/vendor/twig/twig/lib/Twig/Extension/Core.php(1619): craft\fields\Matrix->getSettingsHtml()
#14 src/vendor/craftcms/cms/src/helpers/Template.php(73): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig_Source), Object(craft\fields\Matrix), 'getSettingsHtml', Array, 'method', false, false)
#15 src/storage/runtime/compiled_templates/d3/d3fe895f9018c83e4bc949e4f4f269bd571a4bc42b27edb4786b9c6a92a9fb25.php(246): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig_Source), Object(craft\fields\Matrix), 'getSettingsHtml', Array, 'method')
#16 src/vendor/twig/twig/lib/Twig/Template.php(189): __TwigTemplate_c44cdcf674f7b0bf1e36098bf04e88b1f6ac364281c31f1c632299fe4044e021->block_content(Array, Array)
#17 src/storage/runtime/compiled_templates/6d/6d2df3e35feb7cab6c2821a0f532077c601a5250585109fa9753fec1a66dd157.php(639): Twig_Template->displayBlock('content', Array, Array)
#18 src/vendor/twig/twig/lib/Twig/Template.php(189): __TwigTemplate_24f570570853fa62a7024333fdcab904c71a3bf703e761cad09dc9349307e3d3->block_main(Array, Array)
#19 src/storage/runtime/compiled_templates/6d/6d2df3e35feb7cab6c2821a0f532077c601a5250585109fa9753fec1a66dd157.php(536): Twig_Template->displayBlock('main', Array, Array)
#20 src/vendor/twig/twig/lib/Twig/Template.php(189): __TwigTemplate_24f570570853fa62a7024333fdcab904c71a3bf703e761cad09dc9349307e3d3->block_body(Array, Array)
#21 src/storage/runtime/compiled_templates/5b/5b56439f11bd755a1e9b4a5397ee9f3008177aad0fd4949b7e5b3c14938e2dd1.php(60): Twig_Template->displayBlock('body', Array, Array)
#22 src/vendor/twig/twig/lib/Twig/Template.php(390): __TwigTemplate_12ba38bebbcd85e5c5df943ec3d1552b518f792ff196c5f1f5781a7a145bbdc4->doDisplay(Array, Array)
#23 src/vendor/craftcms/cms/src/web/twig/Template.php(49): Twig_Template->displayWithErrorHandling(Array, Array)
#24 src/vendor/twig/twig/lib/Twig/Template.php(367): craft\web\twig\Template->displayWithErrorHandling(Array, Array)
#25 src/vendor/craftcms/cms/src/web/twig/Template.php(31): Twig_Template->display(Array, Array)
#26 src/storage/runtime/compiled_templates/b5/b5444f45b8ac16412e202937f35913b5a8e68929120674dc4b7ca4123edc7c8e.php(40): craft\web\twig\Template->display(Array, Array)
#27 src/vendor/twig/twig/lib/Twig/Template.php(390): __TwigTemplate_80611249ccffa39a734f2d73d748582e0aae2aba3c06376beb3f7b52d83faadc->doDisplay(Array, Array)
#28 src/vendor/craftcms/cms/src/web/twig/Template.php(49): Twig_Template->displayWithErrorHandling(Array, Array)
#29 src/vendor/twig/twig/lib/Twig/Template.php(367): craft\web\twig\Template->displayWithErrorHandling(Array, Array)
#30 src/vendor/craftcms/cms/src/web/twig/Template.php(31): Twig_Template->display(Array, Array)
#31 src/storage/runtime/compiled_templates/6d/6d2df3e35feb7cab6c2821a0f532077c601a5250585109fa9753fec1a66dd157.php(94): craft\web\twig\Template->display(Array, Array)
#32 src/vendor/twig/twig/lib/Twig/Template.php(390): __TwigTemplate_24f570570853fa62a7024333fdcab904c71a3bf703e761cad09dc9349307e3d3->doDisplay(Array, Array)
#33 src/vendor/craftcms/cms/src/web/twig/Template.php(49): Twig_Template->displayWithErrorHandling(Array, Array)
#34 src/vendor/twig/twig/lib/Twig/Template.php(367): craft\web\twig\Template->displayWithErrorHandling(Array, Array)
#35 src/vendor/craftcms/cms/src/web/twig/Template.php(31): Twig_Template->display(Array, Array)
#36 src/storage/runtime/compiled_templates/d3/d3fe895f9018c83e4bc949e4f4f269bd571a4bc42b27edb4786b9c6a92a9fb25.php(95): craft\web\twig\Template->display(Array, Array)
#37 src/vendor/twig/twig/lib/Twig/Template.php(390): __TwigTemplate_c44cdcf674f7b0bf1e36098bf04e88b1f6ac364281c31f1c632299fe4044e021->doDisplay(Array, Array)
#38 src/vendor/craftcms/cms/src/web/twig/Template.php(49): Twig_Template->displayWithErrorHandling(Array, Array)
#39 src/vendor/twig/twig/lib/Twig/Template.php(367): craft\web\twig\Template->displayWithErrorHandling(Array, Array)
#40 src/vendor/craftcms/cms/src/web/twig/Template.php(31): Twig_Template->display(Array, Array)
#41 src/vendor/twig/twig/lib/Twig/Template.php(375): craft\web\twig\Template->display(Array)
#42 src/vendor/twig/twig/lib/Twig/Environment.php(289): Twig_Template->render(Array)
#43 src/vendor/craftcms/cms/src/web/View.php(331): Twig_Environment->render('settings/fields...', Array)
#44 src/vendor/craftcms/cms/src/web/View.php(378): craft\web\View->renderTemplate('settings/fields...', Array)
#45 src/vendor/craftcms/cms/src/web/Controller.php(155): craft\web\View->renderPageTemplate('settings/fields...', Array)
#46 src/vendor/craftcms/cms/src/controllers/FieldsController.php(243): craft\web\Controller->renderTemplate('settings/fields...', Array)
#47 [internal function]: craft\controllers\FieldsController->actionEditField(71, Object(craft\fields\Matrix), '1')
#48 src/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#49 src/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#50 src/vendor/craftcms/cms/src/web/Controller.php(103): yii\base\Controller->runAction('edit-field', Array)
#51 src/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('edit-field', Array)
#52 src/vendor/craftcms/cms/src/web/Application.php(282): yii\base\Module->runAction('fields/edit-fie...', Array)
#53 src/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('fields/edit-fie...', Array)
#54 src/vendor/craftcms/cms/src/web/Application.php(271): yii\web\Application->handleRequest(Object(craft\web\Request))
#55 src/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#56 src/web/index.php(42): yii\base\Application->run()
#57 {main}
```

**Comment by sergeifilippov** [comment]
I've created a [ticket here](https://github.com/verbb/super-table/issues/200).

**Comment by sergeifilippov** [comment]
## Sidenote

I was able to "bypass" the error and save the change to the field settings by setting my env to a non `local` value.

However, I found that all of the sections using this field lost their data associated with the update field.

**Comment by brandonkelly** [comment]
See verbb/super-table#194 for the original issue.

**Comment by brandonkelly** [comment]
@sergeifilippov Yikes... I hope you have a database backup from before you saved the field!

**Comment by sergeifilippov** [comment]
Yeah well good thing I always test on local.

This particular field is only used on the **Homepage**.  I had another older version open in another tab, saving it restored the content.

I have updated the module using the method defined in your [fix](https://github.com/verbb/super-table/issues/194) and the Twig error is one. Yay!

But data still vanishes on save.....

**Comment by brandonkelly** [comment]
@sergeifilippov Sorry, there was a bug in 3.0.23 where each time you saved a Matrix field, it would geta  new content table 😱

Just released 3.0.23.1 with a fix. If you are able to revert to an older DB backup, great. Otherwise you can send a DB backup into support@craftcms.com and we can patch it up for you.

**Comment by sergeifilippov** [comment]
Well that's 2-3 hours I will never get back. Oh well. You live you learn :D
